src/http/status.rs

```diff
@@ -51,328 +51,215 @@ async fn handle_metrics(
-         # TYPE routinator_valid_roas gauge"
-    ).unwrap();
-    for tal in &metrics.tals {
-        writeln!(res,
-            "routinator_valid_roas{{tal=\"{}\"}} {}",
-            tal.name(), tal.publication.valid_roas
-        ).unwrap();
-    }
-
-    // vrps_total
-    writeln!(res,
-        "\n\
-         # HELP routinator_vrps_total number of valid VRPs per TAL\n\
-         # TYPE routinator_vrps_total gauge"
-    ).unwrap();
-    for tal in &metrics.tals {
-        writeln!(res,
-            "routinator_vrps_total{{tal=\"{}\"}} {}",
-            tal.name(), tal.vrps.valid
-        ).unwrap();
-    }
 
-    // vrps_final
-    writeln!(res,
-        "\n\
-        # HELP routinator_vrps_final final number of valid VRPs\n\
-        # TYPE routinator_vrps_final gauge\n\
-        routinator_vrps_final {}",
-        metrics.vrps.contributed,
-    ).unwrap();
+    let mut res = PrometheusMetrics::new();
 
-    // vrps_unsafe
-    writeln!(res,
-        "\n\
-         # HELP routinator_vrps_unsafe \
-                VRPs overlapping with rejected CAs\n\
-         # TYPE routinator_vrps_unsafe gauge"
-    ).unwrap();
+    // Per-TAL metrics.
+    //
+    // For historical reasons, they don’t include a "_tal_" portion in the
+    // name.
+    res.publication_headers("per TAL");
+    res.vrp_headers("per TAL");
     for tal in &metrics.tals {
-        writeln!(res,
-            "routinator_vrps_unsafe{{tal=\"{}\"}} {}",
-            tal.name(), tal.vrps.marked_unsafe
-        ).unwrap();
+        res.publication_values("tal", &tal.name(), &tal.publication);
+        res.vrp_values("tal", &tal.name(), &tal.vrps);
     }
 
-    // vrps_filtered_locally
-    writeln!(res,
-        "\n\
-         # HELP routinator_vrps_filtered_locally \
-                VRPs filtered based on local exceptions\n\
-         # TYPE routinator_vrps_filtered_locally gauge"
-    ).unwrap();
-    for tal in &metrics.tals {
-        writeln!(res,
-            "routinator_vrps_filtered_locally{{tal=\"{}\"}} {}",
-            tal.name(), tal.vrps.locally_filtered
-        ).unwrap();
-    }
+    res.header(
+        "vrps_added_locally", "gauge",
+        "VRPs added from local exceptions"
+    );
+    res.single("vrps_added_locally", metrics.local.contributed);
 
-    // vrps_duplicate
-    writeln!(res,
-        "\n\
-         # HELP routinator_vrps_duplicate number of duplicate VRPs per TAL\n\
-         # TYPE routinator_vrps_duplicate gauge"
-    ).unwrap();
-    for tal in &metrics.tals {
-        writeln!(res,
-            "routinator_vrps_duplicate{{tal=\"{}\"}} {}",
-            tal.name(), tal.vrps.duplicate
-        ).unwrap();
+    res.header(
+        "stale_objects", "gauge",
+        "total number of stale manifests and CRLs"
+    );
+    res.single("stale_objects", metrics.publication.stale_objects());
+
+    // Per-repository metrics.
+    res.set_prefix("routinator_repo");
+    res.set_prefix("routinator");
```

Comment:
and this is the default value set by `PrometheusMetrics::new`, so this one can go as well

Reply:
This was intended to be _after_ the pre-repository metrics to reset the prefix to its default. This way `publication_values` and `vrp_values` methods can be used both for the TA and repository metrics.